universum/modules/artifact_collector.py

```diff
@@ -175,6 +182,28 @@ def set_and_clean_artifacts(self, project_configs: Configuration, ignore_existin
             with self.structure.block(block_name=name, pass_errors=True):
                 self.preprocess_artifact_list(report_artifact_list, ignore_existing_artifacts)
 
+    def get_conditional_step_branches_artifacts(self, configuration: Step) -> List[ArtifactInfo]:
+        succeeded_config_artifact: Optional[ArtifactInfo] = self.get_config_artifact_if_exists(configuration.if_succeeded)
+        failed_config_artifact: Optional[ArtifactInfo] = self.get_config_artifact_if_exists(configuration.if_failed)
+
+        if succeeded_config_artifact and failed_config_artifact:
```

Comment:
Please take a look a this code:
```python
result = []
if succeeded_config_artifact: result.append(succeeded_config_artifact)
if failed_config_artifact: result.append(failed_config_artifact)
return result
```

What do you think about its readability?

Reply:
Alternatively, the entire function can be merged into `preprocess_artifact_list` by using similar logic

Reply:
> What do you think about its readability?

Much better, thanks.)